src/services/Functional/User.js

```diff
@@ -0,0 +1,34 @@
+import UserAPIService from "@/services/API/User.js";
+
+export default {
+  reAuthenticate(store) {
+    // return the stored reAuthCall if re-authentication is in process
+    if (store.state.auth.isReAuthenticating)
+      return store.state.auth.reAuthenticationCall;
+
+    // re authentication is in process
+    store.dispatch("auth/setReAuthenticationState", true);
+
+    // start the refreshAccessToken call and save that promise in the variable
```

Comment:
every from "and save" should be written before the call to vuex before `return reAuthenticationCall`